monkey/monkey_island/cc/ui/src/components/ui-components/AdvancedMultiSelect.js

```diff
@@ -162,6 +161,16 @@ class AdvancedMultiSelect extends React.Component {
     this.onChange(safePluginNames);
   }
 
+  static getDerivedStateFromProps(props, current_state) {
+    let propsPluginNames = props.value.map(v => v.name);
+    if (current_state.selectedPluginNames !== propsPluginNames) {
+      return {
+      selectedPluginNames: propsPluginNames
+      }
+    }
+    return null
+  }
+
```

Comment:
I couldn't find any good examples of this. Their suggestion is to not use state since we're just reformatting the props in the state. That's the ideal solution, but it would also require a bigger refactoring.